### DISCUSSION TITLE
Don't assign a value to target if it has not changed ? (for Realm db live objects)

### DISCUSSION THREAD

**Discussion by softlion** [riok/mapperly#269] [General]
Hello great devs !

I was using automapper and I'm switching to mapperly as automapper is not correctly supported on iOS devices (runtime code generation fails for slightly "advanced" mappers).

So when using this kind of mapping:
```
    private void ApiObjectToRealmObject(ApiObject source, RealmObject existingTarget) {}
```

Is there a way to instruct the mapper that it should NOT set the target property if its value has not changed ?

With that feature, the set method would not be called and would not trigger a dirty state in the target object (ie: no automatic db update triggered by Realm's live objects feature).

**Comment by latonz** [comment]
Seems related to https://github.com/riok/mapperly/issues/98, would this cover your use cases or do you also need support for non-reference type default values?

**Reply by softlion** [reply]
It's not about null values. It's about same values. So #98 is a corner case of "my" issue, which is more general than decide what to do with null values.

If a target value is "10" and the source is "10", don't assign the target.
In the current version, the target setter is still called.
A new option may be created so that the target value is compared to the source values, and the setter is called only if the value has changed.

**Reply by latonz** [reply]
I understand and I can see the use case of this feature. However, for now other features seem to have higher priority.

**Reply by softlion** [reply]
I agree. But why close it without putting it somewhere else ? That would prevent 1) that someone else open the same issue again. and 2) that this issue is forgotten forever ?

**Reply by latonz** [reply]
A sorry, forgot to reopen. Closed it beforehand since no updates were added to this discussion for over a month. Created https://github.com/riok/mapperly/issues/296.
